### ISSUE TITLE
case functions not properly found after first run

### ISSUE THREAD

**Issue by lawschlosser** [smarie/python-pytest-cases#72]
## Description
Hi,
I believe I've encountered a bug where cases aren't properly identified/used on secondary runs of the test..  In other words, I can run a full pytest invocation as expected (cases work great).  However, if I invoke the same pytest command again, cases are not properly gathered, resulting in skipped tests.

## Repro

```python
# ---- cases.py ----
def case_():
    pass
```

```python
# ---- test.py ----
import pytest_cases
import cases 

@pytest_cases.cases_data(module=cases)
def test_case(case_data):
    pass
```

```bash
pytest ./test.py -p no:warnings
# === 1 passed in 0.01 seconds ====
```

Works as expected. Excellent!

Let's try running it a _second_ time...

```bash
pytest ./test.py -p no:warnings
# == 1 skipped in 0.01 seconds ====
```

oh no :cry:

## Triage
The bug can be remedied by simply deleting the .pyc file for the `cases` module.
```bash
rm cases.pyc
pytest ./test.py -p no:warnings
# === 1 passed in 0.01 seconds ====
```
and again ...
```bash 
rm cases.pyc
pytest ./test.py -p no:warnings
# === 1 passed in 0.01 seconds ====
```

At first I though this was some kind of caching "feature", but ultimately it looks like the bug boils down to this line: https://github.com/smarie/python-pytest-cases/blob/master/pytest_cases/main_params.py#L338
```python
if code.co_filename == module.__file__:
```

The issue is that sometimes `module.__file__` will be the path to the .py file (e.g. `cases.py`) , while in subsequent runs it will be the path to the .pyc file (e.g. `cases.pyc`). So when that path is compared to `code.co_filename` (which _always_ refers the .py file), it will sometimes match and sometimes not.

Thanks for your work!

**Comment by smarie** [comment]
Thanks for reporting ! I'll investigate now

**Comment by smarie** [comment]
I can not reproduce the issue on windows. On which platform are you running this ? The pytest caching mechanism/lib seems to be platform-dependent

**Comment by lawschlosser** [comment]
> I can not reproduce the issue on windows. On which platform are you running this ? The pytest caching mechanism/lib seems to be platform-dependent

@smarie Ah, sorry.  I forgot to mention my platform info:
- ubuntu bionic
- python 2.7.15

libs:
```
pytest==4.6.8
pytest-cases==1.12.0
pytest-mock==2.0.0
```

Thanks for fixing!

**Comment by smarie** [comment]
Thanks for the additional info!
Let me know if the new version does not solve the issue

**Comment by smarie** [comment]
Fixed in 1.12.1 - sorry for the deployment delay, I actually forgot to tag :)
